Adds Collectd CI Package repos

petems opened the pull request:
* Allows installation of 5.6 version of Collectd or master release (potential 5.7)
* Refactor repo management behaviour into it's own classes
* Add parameter to specify version from bleeding edge repos from CI

Closes https://github.com/voxpupuli/puppet-collectd/issues/578

igalic commented on manifests/repo.pp:
```diff
@@ -0,0 +1,38 @@
+          }
+        }
+      }
+      default: {
```
this whole class can be simplified to
```puppet
$os_class = downcase($::operatingsystem)
if defined("::collectd::repo::${os_class}") {
  include "::collectd::repo::$os_class"
} else {
  notify{"You have asked to manage_repo on a ${::osfamily} family that doesn't have a repo specified: ${::operatingsystem}":}
}
```

petems replied:
Nice idea 👍

petems replied:
Done 👍

igalic commented:
some (_related_) tests are failing:

```
    when manage_repo is false
        should not contain Apt::Source[ppa_collectd] (FAILED - 3)
```

jamtur01 commented:
@petems See updates to your branch for fixes to Rubucop fails and added 5.7 support.

petems commented:
@jamtur01 Much appreciated! 👍

jamtur01 commented:
@igalic The test failures seem to be a) network issue and b) deprecation notice from dependency. Not sure either are related to this PR. Any thoughts?

petems commented:
@jamtur01 The tests failures were actually issues, the CentOS code broke the EPEL ordering and the Debian code wasn't idempotent. I fixed the EPEL issue and changed the notify to a warning to fix the idempotency, we're good to go 😄 

@igalic Tests green, let's merge this sucka! 👍

bastelfreak commented on manifests/repo/debian.pp:
```diff
@@ -0,0 +1,31 @@
+  } else {
+    if $::operatingsystem == 'Debian' {
+      warning('Youre trying to use the Ubuntu PPA on a Debian Server, which may cause errors')
+      warning('We recomend you use the $ci_package_repo parameter if you with to use an upstream repo on Debian')
```
" if you with to " sounds like a typo?
